### ISSUE TITLE
Feature request: Option for changing the default indentation of conditional expression consequent/alternate nodes

### ISSUE THREAD

**Issue by apb7** [eslint/eslint#10138]
**Eslint Version:** 4.19.0
**Scenario:**  The `indent` rule creates a new "scope" for ternary operators. For example, if the code is something like this:  
```javascript
textFragments.forEach(function(fragment) {
      result += (
        fragment.type === 'text' ?
        $filter('convertUnicodeToHtml')(fragment.data) :
        '<html-parameter>' + fragment.data +
        '</html-parameter>');
    });
```
Eslint produces an error message: 
```bash
  4:1  error  Expected indentation of 6 spaces but found 4  indent
  5:1  error  Expected indentation of 6 spaces but found 4  indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.
```
On using `--fix`, the code becomes:  
```javascript
textFragments.forEach(function(fragment) {
      result += (
        fragment.type === 'text' ?
          $filter('convertUnicodeToHtml')(fragment.data) :
          '<html-parameter>' + fragment.data +
        '</html-parameter>');
    });
```
Conditional operators, in general, should not create a new scope.  

**Eslint configuration:**
```
{
  "extends": [
    "eslint:recommended",
  ],
  "plugins": [
    "angular",
    "html"
  ],
  "rules": {
    "angular/di-unused": 2,
    "angular/no-inline-template": [
      "error",
      {
        "allowSimple": true
      }
    ],
    "brace-style":[
      "error",
      "1tbs"
    ],
    "camelcase": [
      "error",
      {
        "properties": "never"
      }
    ],
    "comma-spacing": [
      "error",
      {
        "before": false,
        "after": true
      }
    ],
    "curly": [
      "error",
      "all"
    ],
    "dot-notation": [
      "error",
      {
        "allowKeywords": false
      }
    ],
    "eqeqeq": [
      "error",
      "always"
    ],
    "func-style": [
      "error",
      "expression"
    ],
    "indent": [
      "error",
      2,
      {
        "ArrayExpression": 1,
        "flatTernaryExpressions": true,
        "FunctionExpression": {
          "body": 1,
          "parameters": 2
        },
        "MemberExpression": 1,
        "SwitchCase": 1
      }
    ],
    "keyword-spacing": [
      "error",
      {
        "before": true,
        "after": true
      }
    ],
    "max-len": [
      "error",
      "code": 80,
      "ignoreUrls": true
    ],
    "no-compare-neg": "off",
    "no-console": [
      "error",
      {
        "allow": [
          "error",
          "warn"
        ]
      }
    ],
    "no-constant-condition": [
      "off"
    ],
    "no-empty": [
      "error",
      {
        "allowEmptyCatch": true
      }
    ],
    "no-multi-spaces": [
      "error"
    ],
    "no-multi-str": [
      "error"
    ],
    "no-redeclare": [
      "off"
    ],
    "no-useless-escape": "off",
    "no-trailing-spaces": [
      "error"
    ],
    "no-undef": [
      "off"
    ],
    "no-unused-vars": [
      "off"
    ],
    "one-var": [
      "off"
    ],
    "operator-linebreak": [
      "error",
      "after"
    ],
    "padded-blocks": [
      "error",
      "never"
    ],
    "quotes": [
      "error",
      "single",
      {
        "avoidEscape": true
      }
    ],
    "quote-props": [
      "error",
      "as-needed",
      {
        "keywords": true
      }
    ],
    "semi": [
      "error",
      "always"
    ],
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "ignore",
        "asyncArrow": "ignore",
        "named": "never",
      }
    ],
    "space-infix-ops": [
      "error"
    ],
    "space-unary-ops": [
      "error",
      {
        "words": true,
        "nonwords": false,
      }
    ],
    "wrap-iife": [
      "error",
      "any"
    ]
  }
}

```
**Proposed solution:** Provide an option within `indent` rule to turn off this spacing or change the default indentation.

**Comment by apb7** [comment]
Hi @Aladdin-ADD, can you please provide an update on this? Thanks!

**Comment by ilyavolodin** [comment]
Indentation is used for a lot of different reasons, in some cases - to indicate creation of a new "scope" and in some cases just to indicate continuation of multi-line expression. In this case, ternary expression spans multiple lines and indentation is there to make the code more readable at a glance. In my opinion, that's much more prevalent style. I can't really think of a code-base that I've worked with that didn't indent multi-line ternary. But it's a matter of preferences.
Problem is, indent rule is really complex and we try to avoid adding new options to it for fear of breaking something.

**Comment by apb7** [comment]
Hi @ilyavolodin, thank you for your response! I understand your concern.
At our organization, [Oppia](http://github.com/oppia/), we earlier used Eslint version 3.18. A few months back, we migrated to version 4.19 and had to adopt the scope creation for ternary operators pattern. So we were looking for an option by which this scope creation can be enabled or disabled according to one's preferences since we do not really indent multi-line ternary operators in the code base.
Thanks!

**Comment by platinumazure** [comment]
@apb7 You could try using the `ignoredNodes` option to avoid enforcing the indent rule on that node type.

**Comment by apb7** [comment]
Hi @platinumazure, thank you! This does fix the issue, closing this. Thanks again!
